diff -urN aa/dia-0.92.2/configure.in dia-0.92.2/configure.in
--- aa/dia-0.92.2/configure.in	2003-11-01 12:23:46.000000000 +0100
+++ dia-0.92.2/configure.in	2004-03-13 00:21:14.104026880 +0100
@@ -69,7 +69,8 @@
     AC_MSG_CHECKING([if FreeType version is 2.0.9 or higher])
     old_CPPFLAGS="$CPPFLAGS"
     CPPFLAGS="$CPPFLAGS `freetype-config --cflags`"
-    AC_TRY_CPP([#include <freetype/freetype.h>
+    AC_TRY_CPP([#include <ft2build.h>
+#include FT_FREETYPE_H
 #if (FREETYPE_MAJOR*1000+FREETYPE_MINOR)*1000+FREETYPE_PATCH < 2000009
 #error Freetype version too low.
 #endif
