--- dia-cvs-snapshot.orig/app/Makefile.am	Tue Apr 23 09:23:06 2002
+++ dia-cvs-snapshot/app/Makefile.am	Fri Apr 26 11:48:03 2002
@@ -229,7 +229,7 @@
 	pixmaps/magnify-minus-data.xbm \
 	pixmaps/magnify-minus-mask.xbm
 
-run_dia.sh: Makefile.am
+run_dia.sh$(EXEEXT): Makefile.am
 	echo "#!$(SHELL) " > run_dia.sh
 	echo "export DIA_LIB_PATH DIA_SHAPE_PATH DIA_INT_SHAPE_PATH DIA_SHEET_PATH" >> run_dia.sh
 	echo DIA_LIB_PATH=`pwd`/../objects//:`pwd`/../plug-ins// >> run_dia.sh
@@ -240,7 +240,7 @@
 	echo "\$$DEBUGGER `pwd`/dia \$$*" >> run_dia.sh
 	chmod a+x run_dia.sh
 
-run_diaconv.sh: run_dia.sh
+run_diaconv.sh$(EXEEXT): run_dia.sh$(EXEEXT)
 	cat run_dia.sh | sed -e "s%`pwd`/dia%`pwd`/diaconv%g" >run_diaconv.sh 
 	chmod a+x run_diaconv.sh
 
