diff -Nur dia-0.88.1.orig/app/Makefile.am dia-0.88.1/app/Makefile.am
--- dia-0.88.1.orig/app/Makefile.am	Wed May  9 06:05:30 2001
+++ dia-0.88.1/app/Makefile.am	Fri Oct 12 15:48:33 2001
@@ -159,7 +159,7 @@
 	$(LIBART_LIBS) $(PNG_LIBS) $(GDK_PIXBUF_LIBS) $(INTLLIBS) \
 	$(UNICODE_LIBS)
 
-LDFLAGS = -export-dynamic
+dia_embedd_LDFLAGS = -export-dynamic
 
 EXTRA_DIST = \
 	GNOME_Dia.oaf \
@@ -192,7 +192,7 @@
 	pixmaps/magnify-minus-data.xbm \
 	pixmaps/magnify-minus-mask.xbm
 
-run_dia.sh: Makefile.am
+run_dia.sh$(EXEEXT): Makefile.am
 	echo "#!$(SHELL) " > run_dia.sh
 	echo "export DIA_LIB_PATH DIA_SHAPE_PATH DIA_INT_SHAPE_PATH DIA_SHEET_PATH" >> run_dia.sh
 	echo DIA_LIB_PATH=`pwd`/../objects/standard/.libs >> run_dia.sh
diff -Nur dia-0.88.1.orig/configure.in dia-0.88.1/configure.in
--- dia-0.88.1.orig/configure.in	Sat May 19 06:49:28 2001
+++ dia-0.88.1/configure.in	Fri Oct 12 15:45:24 2001
@@ -11,7 +11,7 @@
 
 dnl Initialize libtool
 AC_DISABLE_STATIC
-AC_PROG_LIBTOOL
+AM_PROG_LIBTOOL
 
 dnl Build time sanity check
 AM_SANITY_CHECK
