--- dia-cvs-snapshot.orig/lib/Makefile.am	Wed Aug 13 09:23:14 2003
+++ dia-cvs-snapshot/lib/Makefile.am	Fri Aug 15 16:05:46 2003
@@ -133,11 +133,11 @@
 #		charconv.c \
 #		charconv.h 
 #		render.h \
-#		render.c \
+#		render.c
 
 libdia_la_LDFLAGS = -export-dynamic -module -avoid-version
 
-CPPFLAGS = \
+AM_CPPFLAGS = \
 	$(LIBART_CFLAGS) \
 	-DLIBDIR=\"$(libdir)\" \
 	-DDATADIR=\"$(pkgdatadir)\"
--- dia-cvs-snapshot.orig/app/Makefile.am	Mon Jul  7 09:23:09 2003
+++ dia-cvs-snapshot/app/Makefile.am	Fri Aug 15 16:04:55 2003
@@ -182,7 +182,7 @@
 #$(BONOBO_LIBS)
 
 if HAVE_GCC
-LDFLAGS = -export-dynamic
+AM_LDFLAGS = -export-dynamic
 endif
 
 EXTRA_DIST = \
