diff -urN aa/dia-0.92.2/configure.in dia-0.92.2/configure.in
--- aa/dia-0.92.2/configure.in	2003-11-01 12:23:46.000000000 +0100
+++ dia-0.92.2/configure.in	2004-03-13 00:17:14.392468576 +0100
@@ -216,7 +216,7 @@
 GETTEXT_PACKAGE=dia
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE",[Needed for glib gettext])
-ALL_LINGUAS="am az be ca cs da de el en_GB es fi fr ga hu it ja ko mk ml ms nl nn no pl pt pt_BR ro ru sk sl sr sr@Latn sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="am az be ca cs da de el en_GB es fi fr ga hu it ja ko mk ml ms nb nl nn pl pt pt_BR ro ru sk sl sr sr@Latn sv tr uk vi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 dnl INTLOBJS doesn't seem to always get subst'ed, but I believe it's
