--- dia-399526892d86d7e00e2f565e6c50b73c1195c810/meson.build~	2023-09-16 11:51:53.000000000 +0200
+++ dia-399526892d86d7e00e2f565e6c50b73c1195c810/meson.build	2023-09-20 03:02:24.782540486 +0200
@@ -32,7 +32,7 @@
 libpoppler_dep = dependency('poppler', version: '> 0.62.0', required: false)
 libpopplercpp_dep = dependency('poppler-cpp', required: false)
 conf.set('HAVE_POPPLER', libpoppler_dep.found() and libpopplercpp_dep.found())
-libemf_dep = cc.find_library('emf', required: false)
+libemf_dep = cc.find_library('EMF', required: false)
 conf.set('HAVE_LIBEMF', libemf_dep.found())
 libogdf_dep = cc.find_library('ogdf', required: false)
 conf.set('HAVE_OGDF', libogdf_dep.found())
--- dia-399526892d86d7e00e2f565e6c50b73c1195c810/plug-ins/wmf/wmf.cpp.orig	2023-09-20 03:16:33.087448310 +0200
+++ dia-399526892d86d7e00e2f565e6c50b73c1195c810/plug-ins/wmf/wmf.cpp	2023-09-20 03:16:06.856486686 +0200
@@ -60,7 +60,7 @@
 
 #include <pango/pangowin32.h>
 
-#elif HAVE_LIBEMF
+#elif defined(HAVE_LIBEMF)
 /* We have to define STRICT to make libemf/64 work. Otherwise there is
 wmf.cpp:1383:40: error: cast from 'void*' to 'W32::HDC' loses precision
  */
@@ -177,7 +177,7 @@
     W32::HPEN hOldPen;
     if (colour) {
 	W32::COLORREF rgb = W32COLOR(colour);
-#if defined(G_OS_WIN32) || HAVE_LIBEMF
+#if defined(G_OS_WIN32) || defined(HAVE_LIBEMF)
 	if ((renderer->platform_is_nt && renderer->hPrintDC) || renderer->target_emf) {
           W32::LOGBRUSH logbrush;
 	  W32::DWORD    dashes[6];
@@ -255,7 +255,6 @@
 #else
 #  define DIAG_NOTE my_log
 #endif
-#ifdef G_OS_WIN32
 static void
 my_log(WmfRenderer* renderer, const char* format, ...)
 {
@@ -273,7 +272,6 @@
 
     g_free(string);
 }
-#endif
 
 /*
  * renderer interface implementation
@@ -1377,7 +1375,6 @@
   g_object_class_override_property (object_class, PROP_FONT_HEIGHT, "font-height");
 }
 
-#ifdef G_OS_WIN32
 /* plug-in export api */
 static gboolean
 export_data(DiagramData *data, DiaContext *ctx,
@@ -1412,7 +1409,7 @@
     bbox.bottom = (int)((data->extents.bottom - data->extents.top) * scale *
         100 * W32::GetDeviceCaps(refDC, VERTSIZE) / W32::GetDeviceCaps(refDC, VERTRES));
 
-#if HAVE_LIBEMF
+#ifdef HAVE_LIBEMF
     FILE* ofile = g_fopen (filename, "w");
     if (ofile)
       file = CreateEnhMetaFileWithFILEA (refDC, ofile, &bbox, "Created with Dia/libEMF\0");
@@ -1548,6 +1545,7 @@
 };
 
 
+#ifdef G_OS_WIN32
 static DiaObjectChange *
 print_callback (DiagramData *data,
                 const char  *filename,
@@ -1599,7 +1597,7 @@
     filter_register_export(&emf_export_filter);
 
     filter_register_callback (&cb_gdi_print);
-#elif HAVE_LIBEMF
+#elif defined(HAVE_LIBEMF)
     /* not sure if libEMF really saves EMF ;) */
     filter_register_export(&emf_export_filter);
 #endif
